### DISCUSSION TITLE
The RegionsPlugin in not working in React

### DISCUSSION THREAD

**Discussion by ParamPragyan** [katspaugh/wavesurfer.js#3706] [Q&A]
I have tried like multiple ways of importing and adding the regions inside the create instances but it's still not working.
The TimelinePlugin works fine though.

![image](https://github.com/katspaugh/wavesurfer.js/assets/54448525/74f39c3c-8d8f-42aa-9ca0-4e4a35c54edb)

**Comment by katspaugh** [comment]
There's an example with Regions in React here: https://github.com/katspaugh/wavesurfer.js/discussions/3452#discussioncomment-8510231

**Reply by ParamPragyan** [reply]
I have gone through the example but I didn't find a solution to this. As you see TimelinePlugin is working just fine but the region is not even implementing. I also checked the tools and it shows no error in the console and in the inspect section the region doesn't even exist.

**Reply by katspaugh** [reply]
```
wavesurferRef.current.addRegion -> regionsPlugin.addRegion
```

**Reply by ParamPragyan** [reply]
I understand what you trying to say, but you see I am rendering a video so I need to create an instance `videoRef` through `useEffect()` and Hooks can only be called inside of the body of a function component so I can't use `useMemo()` inside `useEffect`.
